### PR TITLE
Isolate WAN clusters in tests

### PR DESCRIPTION
**Pull request by blazember**
WAN tests use the mock network that has `MockJoiner` joiner
implementation that tries to join all members created with the same
factory. This change extends the functionality of `MockJoiner` with
support for cluster isolation, specifically for WAN tests. If isolation
is enabled by a test - disabled by default, no tests should be affected
by this change apart from WAN tests - `MockJoiner` prevents members
outside of a port range from attempting to join to each other. With
other words, this change makes `MockJoiner` to behave like `TcpJoiner`
that looks up for possible cluster members within a limited port range
only.

Attempting to join members of WAN source and target clusters may fail
WAN sync and consistency check tests in edge cases. If there is an
ongoing join request - that establishes a connection - when the sync or
consistency check request is made, the request will find a connection
between the parties and will try to perform the requested action over
that connection. But the connection will be closed once the two members
realize they are members of two separate clusters. This closes the
connection under the ongoing WAN sync/consistency check operation and
fails the test.

Fixes:
https://github.com/hazelcast/hazelcast-enterprise/issues/3180
https://github.com/hazelcast/hazelcast-enterprise/issues/4037
https://github.com/hazelcast/hazelcast-enterprise/issues/4038
https://github.com/hazelcast/hazelcast-enterprise/issues/4072
https://github.com/hazelcast/hazelcast-enterprise/issues/4086
https://github.com/hazelcast/hazelcast-enterprise/issues/4087
https://github.com/hazelcast/hazelcast-enterprise/issues/4088
https://github.com/hazelcast/hazelcast-enterprise/issues/4106

EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/4173

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockJoiner.java [hunk 1]
```diff
@@ -149,12 +157,24 @@ private Node findAliveNode() {
                 continue;
             }
 
+            if (shouldIsolateClusters && isMemberOfIsolatedCluster(foundNode)) {
+                logger.fine("Node for " + address + " is outside of the join range and should not be joined.");
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockJoiner.java:161]: I'm thinking if this should be a warn instead if the two members' configured ports (`config.network.port`) are the same.

**Review by mmedenjak — APPROVED** [review]

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockJoiner.java (hunk 1 above)
Comment (mmedenjak) [reply hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockJoiner.java:161]: Yes, feel free to change it to a clearer message, feels like if we start getting some test failures because of this, it would be good if it was printed in the default logs + it's only for tests anyway.

**Comment by mmedenjak** [comment]
run-lab-run
